zkevm-circuits/src/tx_circuit.rs

```diff
@@ -139,6 +445,559 @@ impl<F: Field> TxCircuitConfig<F> {
+            ]);
+            vec![
+                meta.query_advice(tx_table.tx_id, Rotation::cur()),
+                RlpTxTag::To.expr(),
+                1.expr(), // tag_index == 1
+                meta.query_advice(tx_table.value, Rotation::cur()),
+                RlpDataType::TxSign.expr(),
+            ]
+            .into_iter()
+            .zip(rlp_table.table_exprs(meta).into_iter())
+            .map(|(arg, table)| (enable.clone() * arg, table))
+            .collect()
+        });
+        meta.lookup_any("tx callee address in RLPTable::TxHash", |meta| {
+            let enable = and::expr(vec![
+                meta.query_fixed(q_enable, Rotation::cur()),
+                tag.value_equals(TxFieldTag::CalleeAddress, Rotation::cur())(meta),
+            ]);
+            vec![
+                meta.query_advice(tx_table.tx_id, Rotation::cur()),
+                RlpTxTag::To.expr(),
+                1.expr(), // tag_index == 1
+                meta.query_advice(tx_table.value, Rotation::cur()),
+                RlpDataType::TxHash.expr(),
+            ]
+            .into_iter()
+            .zip(rlp_table.table_exprs(meta).into_iter())
+            .map(|(arg, table)| (enable.clone() * arg, table))
+            .collect()
+        });
+
+        // lookup tx rlc(value).
+        meta.lookup_any("tx rlc(value) in RLPTable::TxSign", |meta| {
+            let enable = and::expr(vec![
+                meta.query_fixed(q_enable, Rotation::cur()),
+                tag.value_equals(TxFieldTag::Value, Rotation::cur())(meta),
+            ]);
+            vec![
+                meta.query_advice(tx_table.tx_id, Rotation::cur()),
+                RlpTxTag::Value.expr(),
+                1.expr(), // tag_index == 1
+                meta.query_advice(tx_table.value, Rotation::cur()),
+                RlpDataType::TxSign.expr(),
+            ]
+            .into_iter()
+            .zip(rlp_table.table_exprs(meta).into_iter())
+            .map(|(arg, table)| (enable.clone() * arg, table))
+            .collect()
+        });
+        meta.lookup_any("tx rlc(value) in RLPTable::TxHash", |meta| {
+            let enable = and::expr(vec![
+                meta.query_fixed(q_enable, Rotation::cur()),
+                tag.value_equals(TxFieldTag::Value, Rotation::cur())(meta),
+            ]);
+            vec![
+                meta.query_advice(tx_table.tx_id, Rotation::cur()),
+                RlpTxTag::Value.expr(),
+                1.expr(), // tag_index == 1
+                meta.query_advice(tx_table.value, Rotation::cur()),
+                RlpDataType::TxHash.expr(),
+            ]
+            .into_iter()
+            .zip(rlp_table.table_exprs(meta).into_iter())
+            .map(|(arg, table)| (enable.clone() * arg, table))
+            .collect()
+        });
+
+        // lookup to check CallDataLength of the tx's call data.
+        meta.lookup_any("tx calldatalength in TxTable", |meta| {
```

Comment:
We can actually lookup to rlp table for tag = `DataPrefix` to get the call data length using input expr `(tx_id, tag = DataPrefix, tag_rindex = 1, value_acc = tx_table.value, type=TxSign)` where current tag is `CallDataLength`.